projects/addon-table/components/table-pagination/table-pagination.component.ts

```diff
@@ -93,12 +102,34 @@ export class TuiTablePaginationComponent {
         return Math.min(this.start + this.size, this.total);
     }
 
+    /**
+     * @deprecated Use `previousPageDisabled` instead.
+     */
     protected get leftDisabled(): boolean {
-        return !this.start;
+        return this.previousPageDisabled;
     }
 
+    /**
+     * @deprecated Use `nextPageDisabled` instead.
+     */
     protected get rightDisabled(): boolean {
-        return this.end === this.total;
+        return this.nextPageDisabled;
+    }
+
+    protected get firstPageDisabled(): boolean {
+        return this.previousPage === null;
+    }
+
+    protected get previousPageDisabled(): boolean {
```

Comment:
Let's drop `Page` from all these getters. This is a pagination component, it's implicit that we are talking about pages. And class prop names are not minified by compiler so it's better to keep them sort bundlesize-wise.